torch/_inductor/compile_fx.py

```diff
@@ -162,6 +162,11 @@ def compile_fx_inner(
         cudagraphs = config.triton.cudagraphs
 
     shape_env = _shape_env_from_inputs(example_inputs)
+
+    graph_name = torch._functorch.aot_autograd.get_graph_being_compiled()
+    graph_no_id_suffix = graph_name[0 : -(len(str(graph_id)) + 1)]
+    is_inference = graph_no_id_suffix[-(len("inference")) :] == "inference"
```

Comment:
No. Figure out a better way to do this

Reply:
It's a breaking change for all of our backends 😩 but sure

Reply:
I think I'll have aot autograd take in an inference backend and default it to fwd_compiler, if that makes sense to you ? cc @Chillee

Reply:
I think you should just stuff it in TracingContext in the (somewhat misnamed) torch._guards

Reply:
cc @voznesenskym

Reply:
Adapted an inference_compiler option.